x/cdp/keeper/cdp.go

```diff
@@ -341,146 +335,123 @@ func (k Keeper) SetGovDenom(ctx sdk.Context, denom string) {
-		}
-		if dc.Amount.LT(dp.DebtFloor) {
-			return sdkerrors.Wrapf(types.ErrBelowDebtFloor, "proposed %s < minimum %s", dc, dp.DebtFloor)
-		}
+func (k Keeper) ValidatePrincipalAdd(ctx sdk.Context, principal sdk.Coin) error {
+	dp, found := k.GetDebtParam(ctx, principal.Denom)
+	if !found {
+		return sdkerrors.Wrap(types.ErrDebtNotSupported, principal.Denom)
+	}
+	if principal.Amount.LT(dp.DebtFloor) {
+		return sdkerrors.Wrapf(types.ErrBelowDebtFloor, "proposed %s < minimum %s", principal, dp.DebtFloor)
 	}
 	return nil
 }
 
 // ValidatePrincipalDraw validates that an asset is valid for use as debt when drawing debt off an existing cdp
-func (k Keeper) ValidatePrincipalDraw(ctx sdk.Context, principal sdk.Coins) error {
-	for _, dc := range principal {
-		_, found := k.GetDebtParam(ctx, dc.Denom)
-		if !found {
-			return sdkerrors.Wrap(types.ErrDebtNotSupported, dc.Denom)
-		}
+func (k Keeper) ValidatePrincipalDraw(ctx sdk.Context, principal sdk.Coin, expectedDenom string) error {
+	if principal.Denom != expectedDenom {
+		return sdkerrors.Wrapf(types.ErrInvalidDebtRequest, "proposed %s, expected %s", principal.Denom, expectedDenom)
+	}
+	_, found := k.GetDebtParam(ctx, principal.Denom)
+	if !found {
+		return sdkerrors.Wrap(types.ErrDebtNotSupported, principal.Denom)
 	}
 	return nil
 }
 
 // ValidateDebtLimit validates that the input debt amount does not exceed the global debt limit or the debt limit for that collateral
-func (k Keeper) ValidateDebtLimit(ctx sdk.Context, collateralDenom string, principal sdk.Coins) error {
+func (k Keeper) ValidateDebtLimit(ctx sdk.Context, collateralDenom string, principal sdk.Coin) error {
 	cp, found := k.GetCollateral(ctx, collateralDenom)
 	if !found {
 		return sdkerrors.Wrap(types.ErrCollateralNotSupported, collateralDenom)
 	}
-
-	for _, dc := range principal {
-		totalPrincipal := k.GetTotalPrincipal(ctx, collateralDenom, dc.Denom).Add(dc.Amount)
-		collateralLimit := cp.DebtLimit.AmountOf(dc.Denom)
-		if totalPrincipal.GT(collateralLimit) {
-			return sdkerrors.Wrapf(types.ErrExceedsDebtLimit, "debt increase %s > collateral debt limit %s", sdk.NewCoins(sdk.NewCoin(dc.Denom, totalPrincipal)), sdk.NewCoins(sdk.NewCoin(dc.Denom, collateralLimit)))
-		}
-		globalLimit := k.GetParams(ctx).GlobalDebtLimit.AmountOf(dc.Denom)
-		if totalPrincipal.GT(globalLimit) {
-			return sdkerrors.Wrapf(types.ErrExceedsDebtLimit, "debt increase %s > global debt limit  %s", sdk.NewCoin(dc.Denom, totalPrincipal), sdk.NewCoin(dc.Denom, globalLimit))
-		}
+	totalPrincipal := k.GetTotalPrincipal(ctx, collateralDenom, principal.Denom).Add(principal.Amount)
+	collateralLimit := cp.DebtLimit.AmountOf(principal.Denom)
+	if totalPrincipal.GT(collateralLimit) {
+		return sdkerrors.Wrapf(types.ErrExceedsDebtLimit, "debt increase %s > collateral debt limit %s", sdk.NewCoins(sdk.NewCoin(principal.Denom, totalPrincipal)), sdk.NewCoins(sdk.NewCoin(principal.Denom, collateralLimit)))
```

Comment:
```suggestion
		return sdkerrors.Wrapf(types.ErrExceedsDebtLimit, "debt increase %s > collateral debt limit %s", sdk.NewCoin(principal.Denom, totalPrincipal), sdk.NewCoin(principal.Denom, collateralLimit))
```

Reply:
Since the error is printing the coins/coin string this doesn't impact output much anyways